src/state-summary/state-card-input_text.html

```diff
@@ -0,0 +1,108 @@
+<link rel="import" href="../../bower_components/polymer/polymer-element.html">
+
+<link rel="import" href="../../bower_components/iron-flex-layout/iron-flex-layout-classes.html">
+
+<link rel="import" href="../../bower_components/paper-input/paper-input.html">
+
+<link rel="import" href="../components/entity/state-info.html">
+
+<link rel="import" href="../util/hass-util.html">
+
+<dom-module id="state-card-input_text">
+  <template>
+    <style is="custom-style" include="iron-flex iron-flex-alignment"></style>
+    <style>
+      paper-input {
+        margin-left: 16px;
+      }
+    </style>
+
+    <div class='horizontal justified layout'>
+      <state-info state-obj="[[stateObj]]" in-dialog='[[inDialog]]'></state-info>
+      <paper-input
+        minlength='[[min]]' maxlength='[[max]]' value='{{value}}' pin
+        disabled='[[disabled]]' auto-validate pattern='[[pattern]]'
```

Comment:
We should only `auto-validate` if a pattern is given?

Reply:
If no pattern is provided, pattern gets set to `'.*'` by default - I suppose the alternative is to replace the whole `auto-validate pattern=....` with a `[[some string property]]` which is either empty or contains the whole `auto-validate pattern=...` but this seemed easier?